PT-23-feature/create-hero-section

Sashko-kapillar opened the pull request:
…ssors/delite themeToggle and MobileMenutoggle
Перевіряйте коментуйте, ПР створено з Jira
Коміти не знаю чи підтягнуло у Jira

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a visually rich and interactive hero section on the homepage, featuring new components for animal selection, district selection, service type selection, weight selection, and a search button.
  - Added layered visual elements including a hero image and decorative background.

- **Improvements**
  - Enhanced header styling with updated layout, spacing, and positioning.
  - Streamlined user action buttons with updated styling and modal handling.
  - Updated navigation links to use client-side routing with improved styling.
  - Replaced static homepage content with the new hero section.
  - Added typed Redux hooks and a Redux slice for improved state management.
  - Updated logo navigation to link to the homepage root.

- **Removals**
  - Deleted placeholder homepage components: BenefitsSection, ServicesSection, SpecialistList, StatisticsBlock, and TopSpecialistsSlider.

- **Style**
  - Minor layout and spacing adjustments for consistency.
  - Added CSS rule to adjust padding of option elements.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Sashko-kapillar committed:
PT-23-create HeroSection and HeroComponents/add 2 img/change icon-scissors/delite themeToggle and MobileMenutoggle

yuliia-shl commented:
У контексті веб-розробки та найменування файлів для зображень, використання великої літери в назвах файлів — це, в принципі, не є найкращою практикою. Краще

-Маленькі літери: Рекомендується використовувати лише маленькі літери, оскільки на деяких операційних системах (наприклад, на Linux) шляхи та імена файлів є чутливими до регістру. Тобто, «Mops-Hero.png» і «mops-hero.png» будуть сприйматися як різні файли.

-Дефіси для розділення слів: Замість великих літер у середині слів, краще використовувати дефіси для розділення. Це полегшує читабельність. Наприклад: mops-hero-2x-min.png.

-Конвенції: Якщо ваш проект слідує певним стандартам (наприклад, вимагає використання великої літери в назвах файлів), тоді можна це застосовувати, але в більшості випадків рекомендовано уникати великих літер в іменах файлів, особливо для зображень.

Отже, згідно загальноприйнятих практик, краще змінити на:
hero-mops-2x-min.png

Це забезпечить кращу сумісність і читабельність у більшості систем та браузерів.

yuliia-shl commented on src/components/Hero/SearchButton.tsx:
```diff
@@ -0,0 +1,11 @@
+  };
+  
+  export default SearchButton;
+  
```
Замість цього компоненту - перевикористати компонент Button

Sashko-kapillar replied:
То на наступному тижні

Sashko-kapillar committed:
chose: delete emptytext/change tag <a> to <Link> in NavFile for Header

Sashko-kapillar committed:
styles: change btn area for the cursor in UserActions and Nav

Sashko-kapillar committed:
style: change z-index for Header and HeroImg

Sashko-kapillar committed:
chore: change name img & path to img/del empty text in HeroSection and file

Sashko-kapillar commented on src/components/Header/ThemeToggle:
Sure, I deleted it

Sashko-kapillar commented on src/components/Hero/AnimalSelector.tsx:
```diff
@@ -0,0 +1,15 @@
+    );
+  };
+  
+  export default AnimalSelector;
```
Sure
but i'll write the markup and styles first

yuliia-shl replied:
@Sashko-kapillar need to add possibility to switch between dogs and cats

Sashko-kapillar commented on src/components/Hero/DistrictSelector.tsx:
```diff
@@ -0,0 +1,13 @@
+    );
+  };
+  
+  export default DistrictSelector;
```
Sure
but i'll write the markup and styles first

Sashko-kapillar commented on src/components/Hero/HeroBlob.tsx:
```diff
@@ -0,0 +1,17 @@
+  const DogWithBlob = () => {
+    return (
+      <div className="relative">
+        <div className="absolute rotate-[145deg] bg-white rounded-[265px] border-[2px] border-fire shadow-2xlz-0" />
```
sure

Sashko-kapillar committed:
style: change slyle in HeroBlob/

Sashko-kapillar commented on src/components/Hero/HeroBlob.tsx:
```diff
@@ -0,0 +1,17 @@
+        />
+      </div>
+    );
+  };
```
sure

Sashko-kapillar commented on src/components/Hero/HeroContent.tsx:
```diff
@@ -0,0 +1,12 @@
+  return (
+    <div className="flex flex-col items-start max-w-xl w-full">
+      <HeroTitle />
+    </div>
```
sure
Very good change

Sashko-kapillar replied:
HeroContent я замінив на HeroTitle

Sashko-kapillar committed:
fix: delete HeroContent file & change Content to Title in HeroSection

Sashko-kapillar commented on src/components/Hero/HeroImage.tsx:
```diff
@@ -0,0 +1,18 @@
+    return (
+      <div className="z-10 w-full max-w-md">
+        <img
+          src="/public/images/Mops-Hero-1x-min.png"
```
sure

Sashko-kapillar commented on src/components/Hero/SearchButton.tsx:
```diff
@@ -0,0 +1,11 @@
+  };
+  
+  export default SearchButton;
+  
```
Sure
but i'll write the markup and styles first

yuliia-shl commented:
![image](https://github.com/user-attachments/assets/2e41e26f-2a06-420d-b8ea-35cde719390d)
Вушко собаки не має перекривати лого - треба вірно зпозиціонувати
"Обрати район має бути правіше"
![image](https://github.com/user-attachments/assets/528ad611-2afe-413b-b22e-a7f4ab58873e)
При hover має бути відсутня рамка та має бути тінь. Також додати стилі на active

yuliia-shl commented on src/components/Hero/AnimalSelector.tsx:
```diff
@@ -0,0 +1,14 @@
+        </button>
+        <button className="flex-1 px-4 py-2 font-semibold text-fire hover:bg-wildSand">
+          Коти
+        </button>
```
Макет ![image](https://github.com/user-attachments/assets/f8fc3ee7-8a2c-4c0c-bff8-8dc43fbd96b1)
Розміри не відповідають макету
Код
![image](https://github.com/user-attachments/assets/f30dc7c2-91a6-46c5-b67e-84eb598f69e0)

Sashko-kapillar replied:
виправив

yuliia-shl commented on src/components/Hero/ServiceTypeSelector.tsx:
```diff
@@ -0,0 +1,29 @@
+                <button
+                    key={service.label}
+                    className="flex flex-col items-center justify-center gap-2 py-2 px-13 border rounded-[16px] border-fire border-[2px] hover:bg-tenn hover:text-alabaster group"
+                >
```
Розміри не зг макету
код 
![image](https://github.com/user-attachments/assets/0c92c6ff-0edd-4ea6-b922-603b4815c989)
Макет 
![image](https://github.com/user-attachments/assets/d2126b25-ee7f-42e2-8838-571eb6ea659a)

yuliia-shl commented on src/components/Hero/ServiceTypeSelector.tsx:
```diff
@@ -0,0 +1,29 @@
+                    <span>
+                        <svg className="h-13 w-12 shrink-0 fill-tenn group-hover:fill-alabaster" aria-hidden="true">
+                            <use href={`/icons.svg#${service.icon}`} />
+                        </svg>
```
![image](https://github.com/user-attachments/assets/d34dae65-123f-4c04-88ac-0dc81c2e1c9f)
Іконка менша ніж на макеті

Sashko-kapillar replied:
виправив

yuliia-shl commented on src/components/Hero/WeightSelector.tsx:
```diff
@@ -0,0 +1,27 @@
+        {weights.map((w) => (
+          <button
+            key={w.label}
+            className="flex flex-col bg-wildSand px-6 py-3 border-tenn border-[2px] w-34 rounded-[16px] font-medium hover:bg-fire hover:text-white"
```
![image](https://github.com/user-attachments/assets/23e7eb22-9e13-4b09-9b31-c14089d53d68)
Слово не по центру (вилазить) по паддінг більше ніж згідно макету
![image](https://github.com/user-attachments/assets/6c184aef-5d0e-44fb-9093-72b2bb04eab4)
Висота картки теж більша - через що і кнопка буде вища ніж на макеті

Sashko-kapillar replied:
виправив

yuliia-shl commented on src/components/Home/HeroSection.tsx:
```diff
@@ -1,5 +1,24 @@
+      </div>
+    </section>
+  );
 };
```
![image](https://github.com/user-attachments/assets/947041a4-e205-4ba7-9dc0-51dbc5680834)
Це домашня сторінка і має рендеритися на HomePage (route: "/")
Треба виправити структуру, перенести це на HomePage і залитити тільки папку компонента src/components/Hero

Sashko-kapillar replied:
розібрався виправив

Sashko-kapillar commented on src/components/Hero/ServiceTypeSelector.tsx:
```diff
@@ -0,0 +1,29 @@
+                    <span>
+                        <svg className="h-13 w-12 shrink-0 fill-tenn group-hover:fill-alabaster" aria-hidden="true">
+                            <use href={`/icons.svg#${service.icon}`} />
+                        </svg>
```
all right
but i will write the area-label and title for the tooltip highlight

Sashko-kapillar committed:
write the area-label and title for the tooltip highlight on svg

Sashko-kapillar commented on src/components/Hero/ServiceTypeSelector.tsx:
```diff
@@ -0,0 +1,29 @@
+            ))}
+        </div>
+    );
+};
```
i haven't worked on the functionality yet
only created the markup

Sashko-kapillar committed:
chore: change w-34 (isn’t included in Tailwind’s default spacing scale) to [136px]

Sashko-kapillar commented on src/components/Hero/WeightSelector.tsx:
```diff
@@ -0,0 +1,27 @@
+        {weights.map((w) => (
+          <button
+            key={w.label}
+            className="flex flex-col bg-wildSand px-6 py-3 border-tenn border-[2px] w-34 rounded-[16px] font-medium hover:bg-fire hover:text-white"
```
ok

Sashko-kapillar commented on src/components/Hero/WeightSelector.tsx:
```diff
@@ -0,0 +1,27 @@
+      <SearchButton />
+      </div>
+    );
+  };
```
i haven't worked on the functionality yet
only created the markup

petHelpMarketplace commented on src/components/Header/UserActions.tsx:
```diff
@@ -74,7 +10,7 @@ export const UserActions = () => {
   return (
```
Test comment

petHelpMarketplace reviewed (commented):
Finished review

Sashko-kapillar committed:
style: add in HeroContainer justify-between for right position of img

Sashko-kapillar committed:
refactor: delete component Home/move component HeroSection to the correct location (pages/Home/HomePege)/move HeroComponents 9items to the Hero folder/mode MainLayout

Sashko-kapillar committed:
style: aligned the elements according to the layout (district & animal)

Sashko-kapillar commented on src/components/Home/HeroSection.tsx:
```diff
@@ -1,5 +1,26 @@
+      <div className="relative z-20">
+        <HeroFilters />
+      </div>
+    </section>
```
додати ще один <div> ?

Sashko-kapillar replied:
додати ще один тег дів?

Sashko-kapillar commented on src/components/Hero/HeroBlob.tsx:
```diff
@@ -0,0 +1,15 @@
+        <div className="absolute rotate-[145deg] bg-white rounded-[265px] border-[2px] border-fire shadow-2xlz-0" />
+        <img
+          src="/images/blob-behind-dog.png"
+          alt="Dog"
```
ок

Sashko-kapillar committed:
style: switch to px in negative offset

Sashko-kapillar commented on src/components/Hero/HeroBlob.tsx:
```diff
@@ -0,0 +1,15 @@
+        <img
+          src="/images/blob-behind-dog.png"
+          alt="Dog"
+          className="relative z-10 w-[auto] h-[auto] h-auto -top-196 -right-132 "
```
all right

Sashko-kapillar committed:
style: hover border-none in serrviceType
chore: delete empty <span>

Sashko-kapillar committed:
chore: format document

Sashko-kapillar committed:
feat(23): create heroSlice & heroSelectors
style(23): mode btnSelect/HeroImage/HeroBlob/HeroSection/HtroFilters/Header
refactor(23): add the work logic/AnimalSelector/DistrictSelector/STS/
add state & hooks

Sashko-kapillar committed:
style(23): mod style DistrictSelector

Sashko-kapillar committed:
style(23): change padding in the cart to the STS

yuliia-shl commented on src/components/HeroSection/Hero/AnimalSelector.tsx:
```diff
@@ -0,0 +1,33 @@
+  };
+
+  return (
+    <div className="flex justify-center rounded-[12px] overflow-hidden bg-wildSand w-1/2 border-[2px] border-tenn hover:shadow-shark">
+      {(["dog", "cat"] as const).map((animal) => (
+        <button
+          key={animal}
+          onClick={() => handleSelect(animal)}
+          className={`px-4 w-1/2 font-semibold py-2 font-semibold transition  active:shadow-inset-shark ${
+            selectedAnimal === animal
+              ? 'bg-tenn text-alabaster ${animal === "dog" ? "rounded-l-[8px]" : "rounded-r-[8px]"}'
+              : "text-fire"
+          }`}
+        >
+          {animal === "dog" ? "Собаки" : "Коти"}
+        </button>
+      ))}
+    </div>
```
accessibility attributes it is not our goal yet, we will ask you to check later. Don't write about it now

yuliia-shl commented on src/components/HeroSection/Hero/AnimalSelector.tsx:
```diff
@@ -0,0 +1,33 @@
+        <button
+          key={animal}
+          onClick={() => handleSelect(animal)}
+          className={`px-4 w-1/2 font-semibold py-2 font-semibold transition  active:shadow-inset-shark ${
+            selectedAnimal === animal
+              ? 'bg-tenn text-alabaster ${animal === "dog" ? "rounded-l-[8px]" : "rounded-r-[8px]"}'
+              : "text-fire"
+          }`}
```
what exactly preventing?

yuliia-shl commented on src/components/HeroSection/Hero/SearchButton.tsx:
```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react';
+
+const SearchButton = () => {
+  const [openRegisterModal, setOpenRegisterModal] = useState(false);
```
SearchButton should open SeachSpecialistsPage - which is not exist yet - you need to remove this state.
Please delete this component and use <Button/> as reusable component

yuliia-shl commented on src/components/HeroSection/Hero/HeroFilters.tsx:
```diff
@@ -0,0 +1,19 @@
+      <ServiceTypeSelector />
+      <WeightSelector />
+    </div>
+  );
```
![image](https://github.com/user-attachments/assets/3b00f278-5d44-4084-8fca-0fccecb2dad2)
Висота цього контейнеру не відповідає макету

yuliia-shl replied:
його позиціонування також не збігається з макетом
ось що є
![image](https://github.com/user-attachments/assets/0acd3a10-484d-4381-a675-48e0dbe980d3)

А ось макет 
![image](https://github.com/user-attachments/assets/282e9168-dbb3-4c49-a3b8-4f0954639537)
Також відстань між Hero та футером (без врахування блока фільтрів) 159px за макетом - а у тебе 172

yuliia-shl commented on src/components/Header/UserActions.tsx:
```diff
@@ -85,7 +21,7 @@ export const UserActions = () => {
+        className="hover:text-mineShaft transition-colors py-2"
         type="button"
         onClick={() => setOpenRegisterModal(true)}
       >
```
Виправити ховер і фокус (Стати фахівцем)  згідно макету.
![image](https://github.com/user-attachments/assets/5b2c259d-a8b3-40e6-abe8-185bae617502)
EN не має змінювати колір зараз, поки не додамо переклад - кнопки мови не активні

yuliia-shl replied:
висота хедера зідно макету 68 а не 64
![image](https://github.com/user-attachments/assets/52982198-e271-47a1-84cd-4a7662edc743)

yuliia-shl replied:
Також краще розшищити зону клікабельності для кращогоuser experience на всю висотухедера
![image](https://github.com/user-attachments/assets/b14fd3d7-606d-443a-9b38-38c2cb02ae8b)

yuliia-shl commented on src/components/HeroSection/Hero/HeroBlob.tsx:
```diff
@@ -0,0 +1,15 @@
+  };
+  
+  export default DogWithBlob;
+  
```
![image](https://github.com/user-attachments/assets/a2698f60-7126-4087-a597-138bbcab5953)
не знаю що за блок - але він не дає натиснути на лого на футері - це блок його блокує

yuliia-shl commented on src/components/HeroSection/Hero/ServiceTypeSelector.tsx:
```diff
@@ -0,0 +1,46 @@
+          >
+            <title>{service.label}</title>
+            <use href={`/icons.svg#${service.icon}`} />
+          </svg>
```
![image](https://github.com/user-attachments/assets/e1ad972d-1ec6-4100-a272-cc1a310c850a)
Розмір не відповідає макету

yuliia-shl commented on src/components/HeroSection/Hero/WeightSelector.tsx:
```diff
@@ -0,0 +1,27 @@
+        {weights.map((w) => (
+          <button
+            key={w.label}
+            className="flex flex-col gap-[2px] bg-wildSand px-[20px] py-[9px] border-tenn border-[2px] w-[136px] rounded-[16px] font-medium hover:bg-fire hover:text-white hover:shadow-shark active:shadow-inset-shark"
```
![image](https://github.com/user-attachments/assets/65dc96af-fca6-4941-8663-0a132fb2decd)
hover:shadow-shark - колір не по макету і рамки не має бути при ховері

yuliia-shl commented on src/components/HeroSection/Hero/WeightSelector.tsx:
```diff
@@ -0,0 +1,27 @@
+             <span className="text-sm">({w.range})</span>
+          </button>
+        ))}        
+      <SearchButton />
```
Замінити SearchButton  на компонент Button

Sashko-kapillar committed:
chore(23): add aria-label to btn/correct in Serch/del empty div in HeroBlob/add tilda in AnimalSelector

Sashko-kapillar committed:
feat(23): add new logic in SerchBtn

yuliia-shl commented on src/components/HeroSection/Hero/AnimalSelector.tsx:
```diff
@@ -0,0 +1,33 @@
+          key={animal}
+          onClick={() => handleSelect(animal)}
+          className={`px-4 w-1/2 font-semibold py-2 font-semibold transition  active:shadow-inset-shark ${
+            selectedAnimal === animal
```
видалити 1font-semibold - повторюється

yuliia-shl commented on src/components/HeroSection/Hero/ServiceTypeSelector.tsx:
```diff
@@ -0,0 +1,46 @@
+        <button
+          key={service.label}
+          className="flex flex-col items-center justify-center gap-2 py-3 px-13 border rounded-[16px] border-tenn border-[2px] hover:bg-tenn hover:text-alabaster hover:border-none hover:shadow-shark active:shadow-inset-shark group"
+        >
```
'border' applies the same CSS properties as 'border-[2px]'

Sashko-kapillar committed:
style(23): change bg-color to WeightSelector/add full area in nav at header/add hover in nav & user element/mod full height to header 68/add hover in District

Sashko-kapillar committed:
chore(23): del empty el

Sashko-kapillar committed:
style(23): change icon cat-house

Sashko-kapillar committed:
chore(23): add logic to cat switch remove the WeightSelector

Sashko-kapillar committed:
Merge branch 'dev' of https://github.com/petHelpMarketplace/frontend into PT-23-feature/create-hero-section

Sashko-kapillar committed:
chore(23): mode path to RegisterForm

Sashko-kapillar committed:
style(23): change cat-house icon

yuliia-shl commented on src/components/HeroSection/HeroSection.tsx:
```diff
@@ -0,0 +1,20 @@
+const HeroSection = () => {
+  return (
+    <section className="py-12 bg-gradient-to-bl from-red-beech via-tenn to-red-beech w-full rounded-b-[40px] max-w-[1280px] mx-auto -mt-20 px-30 max-h-165 mb-43">
+      <div className="container relative z-10 flex flex-col-reverse lg:flex-row items-center justify-between">
```
mb-43" - замінити на mb-40"

yuliia-shl commented on src/components/HeroSection/Hero/HeroFilters.tsx:
```diff
@@ -0,0 +1,26 @@
+const HeroFilters = () => {
+  const selectedAnimal = useAppSelector(selectAnimal);
+  return (
+    <div className="relative bg-alabaster shadow-md rounded-[20px] py-4 px-8 w-full flex flex-col gap-4 -mt-208 z-10">
```
Замінити -mt-208 на -mt-212

yuliia-shl commented on src/components/HeroSection/Hero/AnimalSelector.tsx:
```diff
@@ -0,0 +1,33 @@
+        <button
+          key={animal}
+          onClick={() => handleSelect(animal)}
+          className={`px-4 w-1/2 font-semibold py-2 font-semibold transition  active:shadow-inset-shark ${
```
замінити py-2  на py-3 - бо висота кнопки має бути 48px а зараз 44px

yuliia-shl commented on src/components/HeroSection/Hero/AnimalSelector.tsx:
```diff
@@ -0,0 +1,33 @@
+  };
+
+  return (
+    <div className="flex justify-center rounded-[12px] overflow-hidden bg-wildSand w-1/2 border-[2px] border-tenn hover:shadow-shark">
```
hover:shadow-shark" - перенести з цього div на кнопки, що нище

yuliia-shl commented on src/components/Header/UserActions.tsx:
```diff
@@ -72,9 +72,9 @@ export const UserActions = () => {
   const [openLoginModal, setOpenLoginModal] = useState(false);
```
Видалити закоментований код

Sashko-kapillar committed:
chore(pt-23): clear the garbage

Sashko-kapillar committed:
style(PT-23): change padding +4px to button in AnimalSelector

Sashko-kapillar committed:
style(PT-23): + 2px to button in District

Sashko-kapillar committed:
chore(PT-23): add 1px to gap in HeroSection

Sashko-kapillar committed:
style(pt-23): change shadow to the AnimalSelector

yuliia-shl reviewed (approved):
Looks good to me
